dapps/src/apps/fetcher.rs

```diff
@@ -155,6 +158,45 @@ impl From<zip::result::ZipError> for ValidationError {
 	}
 }
 
+
+fn sha3(file: &mut fs::File) -> Result<H256, io::Error> {
```

Comment:
yes, but better would be a generic implementation for any `R: Read` so that you can sha3 a buffered reader as well as reading from a raw fd (which is a lot less efficient in general)